src/uwtools/drivers/forecast.py

```diff
@@ -77,14 +82,20 @@ def create_directory_structure(run_directory: DefinitePath, exist_act: str = "de
 
         # Delete or rename directory if it exists.
 
-        handle_existing(str(run_directory), exist_act)
+        if dry_run and os.path.isdir(run_directory):
+            log.info(f"Would {exist_act} directory")
+        else:
+            handle_existing(str(run_directory), exist_act)
 
         # Create new run directory with two required subdirectories.
 
         for subdir in ("INPUT", "RESTART"):
             path = os.path.join(run_directory, subdir)
-            log.info("Creating directory: %s", path)
-            os.makedirs(path)
+            if dry_run:
+                log.info("Would create directory: %s", path)
+            else:
+                log.info("Creating directory: %s", path)
+                os.makedirs(path)
```

Comment:
nitpick: In `stage_files()`, earlier, we log an after-the-fact message about an action we're taking, while here we log a before-the-fact message. It would probably be a nice design to be consistent about this. I think before-the-fact messaging is helpful because, if the operation fails (e.g. throws and exception that halts the program), we've just logged something about the thing that failed; while an after-the-fact message won't appear at all and the user might be left trying to understand what the program was doing when it failed.

Reply:
I've reordered the log messages in stage_files to make them before-the-fact, but if I've missed something, please let me know.